sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1343,6 +1343,23 @@ void CAFMaker::produce(art::Event& evt) noexcept {
 
   rec.hdr = SRHeader();
 
+  // Get the Process and Cluser number
+  const char *process_str = std::getenv("PROCESS");
+  if (process_str) {
+    try {
+      rec.hdr.proc = std::stoi(process_str);
+    }
+    catch (...) {}
+  }
+
+  const char *cluster_str = std::getenv("CLUSTER");
+  if (cluster_str) {
+    try {
+      rec.hdr.cluster = std::stoi(cluster_str);
+    }
+    catch (...) {}
+  }
+
```

Comment:
I agree that the chances that these variables are set, but are not parseable to integers, and that the difference between storing 0 or NaN(?) in the CAF actually matters to anyone, is far outweighted by the grossness of try-catch blocks, so I would go with Gianlucca's short rewriting with `atoi()`

Reply:
`atoi` does not work here because 0 is a valid value for `PROCESS`. Also, I don't think that `std::logic_error` covers all the exceptions that `stoi` throws.

Reply:
> I don't think that `std::logic_error` covers all the exceptions that `stoi` throws.

It actually does (I checked). Both `std::invalid_argument` and `std::out_of_range` are (i.e. derive) `std::logic_error`s.